### DISCUSSION TITLE
How to get editor content in the context of a form submit

### DISCUSSION THREAD

**Discussion by andersr** [ueberdosis/tiptap#3738] [General Questions]
Hi - your intro example shows how to set up an app with a working TipTap editor.  However, I do not see any instructions for how to actually use the text that is generated, eg in the context of a form submission. There is a method, `editor.getHTML()` that is provided but it is not clear how to call it. Can you provide an example of how to use the editor in the context of a form?  Thanks!

**Comment by samuel-belcastro** [comment]
Dropping this in case anyone stumbles across this question. Not sure if this is the most efficient option, but it's a start.


```ts
import { useEditor, EditorContent } from '@tiptap/react'
import StarterKit from '@tiptap/starter-kit'
import { useState } from 'react'

export default function TipTapEditorForm() {
  const [editorContent, setEditorContent] = useState('');
  const editor = useEditor({
      extensions: [StarterKit],
      onUpdate({ editor }) {
          setEditorContent(JSON.stringify(editor.getJSON()))
      }
  });
  
  return (
      <form>
          <div className="mb-3">
              <label className="form-label">My Editor Form Control</label>
              <textarea value={editorContent} hidden></textarea>
              <EditorContent className='form-control p-3' editor={editor} />
          </div>
      </form>
  )
}

```
